src/iterator.vala

```diff
@@ -58,37 +48,26 @@ public class Vast.Iterator : Object
     public Value
     get_value ()
     {
-        return array.get_value (_cursor);
+        return memory_to_value(get(), array.scalar_type, array.scalar_size);
     }
 
     public void
-    set (void* val)
+    set (void* memory)
     {
-        Memory.copy ((uint8*) array.data.get_data () + _offset, val, array.scalar_size);
+        Memory.copy (get(), memory, array.scalar_size);
```

Comment:
Maybe we could keep an internal accessor to the `Bytes` content that would be inlined. This would give `Iterator` and `MultiIterator` a priviledged access to the data and avoid a lot of unnecessary indirections.